maven/pom.xml

```diff
@@ -29,11 +29,11 @@
 
     <java.target>1.8</java.target>
 
-    <maven-plugin-api.version>3.0</maven-plugin-api.version>
+    <maven-plugin-api.version>3.6.3</maven-plugin-api.version>
```

Comment:
Sadly this can't be 'simply' updated without loosing support for older maven versions. Essentially what this change is saying is that the plugin *only* works with `maven => 3.6.3`. However I think this plugin should continue to work with older maven versions.
I'm also quite surprised that the integration-tests didn't complain about this...mhhh...will look into that.

E.g. when using a local test this yields:
```
$ mvn --version
Apache Maven 3.6.1 (d66c9c0b3152b2e69ee9bac180bb8fcc8e6af555; 2019-04-04T21:00:29+02:00)
Maven home: /usr/share/maven-3.6.1
Java version: 9.0.4, vendor: Oracle Corporation, runtime: /usr/lib/jvm/java-9-oracle
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "4.15.0-99-generic", arch: "amd64", family: "unix"
$ mvn clean install && mvn clean package -Pdemo
ERROR] Failed to execute goal pl.project13.maven:git-commit-id-plugin:4.0.1-SNAPSHOT:revision (get-the-git-infos) on project git-commit-id-plugin: The plugin pl.project13.maven:git-commit-id-plugin:4.0.1-SNAPSHOT requires Maven version [3.6.3,) -> [Help 1]
```

References:
https://github.com/git-commit-id/git-commit-id-maven-plugin/issues/316